### PR TITLE
feat(auth): Add MFA info to UserRecord

### PR DESCRIPTION
**Pull request by jasperkuperus**
### Discussion

Issue: https://github.com/firebase/firebase-admin-go/issues/421

This PR implements a small part of Multi Factor Authentication (MFA). I do realize this isn't a full implementation, unfortunately I don't have time for that. This PR only implements unmarshalling the MFA info from the REST response into the `UserRecord` struct.

For structure and naming within the `UserRecord`, I've decided to be consistent with the Node.js client library, which already has this implemented: https://firebase.google.com/docs/reference/admin/node/admin.auth.UserRecord

### Testing

  * I've updated the existing tests to also incorporate MFA. On the existing `testUser`, I've added MFA info. Next to that, I've introduced a test user without MFA. In the list endpoints, I replaced every third user with a non MFA user, so the list tests both users with and without MFA info available. All unit tests should be OK.
  * Unfortunately, I wasn't able to get the integration tests running locally, so I'm not sure whether I did break something there. I did manage to test the code in an actual project with a small local setup and verified that it works.

### API Changes

  * This PR does not change the public API. The REST API already returns the `mfaInfo` section. This PR merely adds the code to pick it up into Golang structs.

### CODE REVIEW

File: auth/user_mgt.go
Comment (hiranya911) [inline auth/user_mgt.go:60]: This API should closely mirror the type signatures defined for Node.js:

```go
type MultiFactorInfo struct {
  UID string
  DisplayName string
  EnrollmentTimestamp int64
  FactorID string
}
```

You will probably need to define a multiFactorInfoQuery internal type for unmarshalling, and then initialize this type based on that.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Thanks @jasperkuperus for the PR. It looks mostly good. I've suggested some changes to the public API.

@bojeil-google how do you feel about just releasing the changes to `UserRecord` without the support for updating MFA settings?

**Comment by bojeil-google** [comment]
> Thanks @jasperkuperus for the PR. It looks mostly good. I've suggested some changes to the public API.
> 
> @bojeil-google how do you feel about just releasing the changes to `UserRecord` without the support for updating MFA settings?

I think it's fine to do so, as long as we keep track of the missing functionality (updating MFA settings). Ultimately, we want to make sure we have feature parity across all languages.

File: auth/user_mgt.go
Comment (jasperkuperus) [reply auth/user_mgt.go:60]: Ah, I must've halfway through accidentally switched to mapping names to the REST API in stead of the Node.js API. I see. I've had a look at how the Node.js library converts the REST response JSON to the internal structures. I'll fix this early next week!

Comment (jasperkuperus) [reply auth/user_mgt.go:60]: Mental note: https://github.com/firebase/firebase-admin-node/blob/f624b2963900deebed0fdb5cbe66ce7eb0d244b8/src/auth/user-record.ts

Comment (jasperkuperus) [reply auth/user_mgt.go:60]: Instead of what you proposed `multiFactorInfoQuery`, I've sticked with `MultiFactorInfoResponse`, which makes more sense to me and is in line with the Node.js naming.

File: auth/user_mgt.go
Comment (hiranya911) [inline auth/user_mgt.go:61]: Suggest also adding `omitempty` to everything (just for consistency with the rest of this file).

File: testdata/list_users.json
Comment (hiranya911) [inline testdata/list_users.json:77]: Nit: Suggest using `+1234567890` as the phone number and `testuser` as the enrollment ID to make it obvious that they are test data.

File: testdata/get_user.json
Comment (hiranya911) [inline testdata/get_user.json:37]: Here too. Use obviously fake data.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Thanks for making the changes @jasperkuperus. This is starting to shape up nicely. Just a few more changes needed to get the public API sorted out.

File: auth/user_mgt.go
Comment (hiranya911) [inline auth/user_mgt.go:87]: This cannot be `PhoneMultiFactorInfo`. We'd want to support other auth types in the future, and at that point we'll have to break this API. Let's just rename `PhoneMultiFactorInfo` struct to `MultiFactorInfo`. Then we can add new fields to that type when we introduce a new auth factor, and that won't be a breaking change.

**Review by hiranya911 — CHANGES_REQUESTED** [review]

**Review by hiranya911 — APPROVED** [review]
Thanks @jasperkuperus. This LGTM 👍 

I do need to get this API approved through our internal API review process. I'll merge when I have the required approvals.

**Comment by hiranya911** [comment]
API review process for this has been initiated. Should have some updates by next week.

**Comment by hiranya911** [comment]
This API has been approved with some minor changes. The change being removal of the `MultiFactorID` type and the related `Phone` constant (we will use `FactorID string` instead).

@jasperkuperus would you like to push the change to this PR? If you're busy I can merge this as-is, and implement the required changes on top of that.

**Comment by jasperkuperus** [comment]
> @jasperkuperus would you like to push the change to this PR? If you're busy I can merge this as-is, and implement the required changes on top of that.

@hiranya911  It's a small change, but I have a rather busy week for me. If you could be so kind to merge it and implement the change on top of that, it'd be of great help, thanks! 😊

**Pull request merged**
